src/app/ui/components/account/account.card.tsx

```diff
@@ -56,7 +56,27 @@ export function AccountCard({
       <Flex flexDir={{ base: 'column', md: 'row' }} justify="space-between">
         <Box mb="space.05" mt="space.04">
           <SkeletonLoader width="200px" height="38px" isLoading={isLoadingBalance}>
-            <styled.h1 textStyle="heading.02">{balance}</styled.h1>
+            <styled.h1
+              textStyle="heading.02"
+              style={{
+                whiteSpace: 'nowrap',
+                display: 'inline-block',
+                transformOrigin: 'left center',
+                maxWidth: '100%',
+              }}
+              ref={el => {
+                if (el) {
+                  const parentWidth = el.parentElement?.offsetWidth || 0;
+                  const naturalWidth = el.scrollWidth;
+                  const scale = parentWidth / naturalWidth;
+                  const finalScale = Math.max(scale);
+                  el.style.transform = finalScale < 1 ? `scale(${finalScale})` : 'none';
+                  el.style.width = finalScale < 1 ? `${100 / finalScale}%` : 'auto'; // Adjust width to accommodate scaled text
+                }
```

Comment:
This could better be wrapped as a reusable react hook with the ref passed in

Reply:
can't believe that I'd say this, but I think I managed to get that one done